File: app/src/main/java/com/yvrun/officeprocess/dagger/component/AppComponent.java
package com.yvrun.officeprocess.dagger.component;


import com.yvrun.officeprocess.dagger.module.AppModule;
import com.yvrun.officeprocess.dagger.module.HttpModule;
import com.yvrun.officeprocess.dagger.module.NeedInjectModules;
import com.yvrun.officeprocess.global.BaseApplication;

import javax.inject.Singleton;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;


@Singleton
@Component(modules = {AndroidInjectionModule.class, NeedInjectModules.class,AppModule.class, HttpModule.class})
public interface AppComponent extends AndroidInjector<BaseApplication> {

    void inject(BaseApplication application);
}

File: kotlinappother/src/main/java/com/chengzhen/kotlinappother/mvp/dagger/component/AppComponent.kt
package com.chengzhen.kotlinappother.mvp.dagger.component


import com.chengzhen.kotlinappother.base.BaseApplication
import com.chengzhen.kotlinappother.mvp.dagger.module.AppModule
import com.chengzhen.kotlinappother.mvp.dagger.module.HttpModule
import com.chengzhen.kotlinappother.mvp.dagger.module.NeedInjectModules

import javax.inject.Singleton

import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector


@Singleton
@Component(modules = [AndroidInjectionModule::class, NeedInjectModules::class, AppModule::class, HttpModule::class])
interface AppComponent : AndroidInjector<BaseApplication> {

    override fun inject(application: BaseApplication)
}
